instrumentation/netty/netty-4.1/library/src/main/java/io/opentelemetry/instrumentation/netty/v4_1/NettyServerTelemetryBuilder.java

```diff
@@ -8,32 +8,32 @@
 import com.google.errorprone.annotations.CanIgnoreReturnValue;
 import io.netty.handler.codec.http.HttpResponse;
 import io.opentelemetry.api.OpenTelemetry;
+import io.opentelemetry.instrumentation.api.incubator.builder.internal.DefaultHttpServerInstrumenterBuilder;
 import io.opentelemetry.instrumentation.api.semconv.http.HttpServerAttributesExtractorBuilder;
-import io.opentelemetry.instrumentation.api.semconv.http.HttpServerRouteBuilder;
-import io.opentelemetry.instrumentation.api.semconv.http.HttpSpanNameExtractorBuilder;
 import io.opentelemetry.instrumentation.netty.v4.common.HttpRequestAndChannel;
-import io.opentelemetry.instrumentation.netty.v4.common.internal.server.NettyServerInstrumenterFactory;
+import io.opentelemetry.instrumentation.netty.v4.common.internal.server.HttpRequestHeadersGetter;
+import io.opentelemetry.instrumentation.netty.v4.common.internal.server.NettyHttpServerAttributesGetter;
 import io.opentelemetry.instrumentation.netty.v4_1.internal.ProtocolEventHandler;
+import io.opentelemetry.instrumentation.netty.v4_1.internal.server.NettyServerInstrumenterBuilderUtil;
 import java.util.List;
 import java.util.Set;
-import java.util.function.Consumer;
 
 /** A builder of {@link NettyServerTelemetry}. */
 public final class NettyServerTelemetryBuilder {
 
-  private final OpenTelemetry openTelemetry;
+  private final DefaultHttpServerInstrumenterBuilder<HttpRequestAndChannel, HttpResponse> builder;
 
-  private Consumer<HttpServerAttributesExtractorBuilder<HttpRequestAndChannel, HttpResponse>>
-      extractorConfigurer = builder -> {};
-  private Consumer<HttpSpanNameExtractorBuilder<HttpRequestAndChannel>>
-      spanNameExtractorConfigurer = builder -> {};
-  private Consumer<HttpServerRouteBuilder<HttpRequestAndChannel>> httpServerRouteConfigurer =
-      builder -> {};
-  private boolean emitExperimentalHttpServerMetrics = false;
   private boolean emitExperimentalHttpServerEvents = false;
 
+  static {
+    NettyServerInstrumenterBuilderUtil.setBuilderExtractor(NettyServerTelemetryBuilder::getBuilder);
```

Comment:
could use `NettyServerInstrumenterBuilderUtil.setBuilderExtractor(nettyServerTelemetryBuilder -> nettyServerTelemetryBuilder.builder);` so you wouldn't need to add the `getBuilder` method